Makefile

```diff
@@ -148,7 +148,7 @@ compare_memory_proof: $(CAIRO_RS_MEM_PROOF) $(CAIRO_MEM_PROOF)
 	cd tests; ./compare_vm_state.sh memory proof_mode
 
 docs:
-	cargo doc --verbose --release --locked --no-deps
+	RUSTDOCFLAGS="--cfg docsrs" cargo +nightly doc --verbose --release --locked --no-deps --all-features --open
```

Comment:
I know this is not meant to be a definitive version, but it would be a good idea to add a comment mentioning:
1. Why we use nightly for the docs.
2. Which version is expected to stabilize the feature, if already known.

Reply:
I put a screen capture in this comment: https://github.com/lambdaclass/cairo-rs/pull/761#issuecomment-1406494057. 

It allows us to use the `doc_cfg` feature(tracking issue: https://github.com/rust-lang/rust/issues/43781, doc: https://doc.rust-lang.org/unstable-book/language-features/doc-cfg.html), and therefore to document optional features in an user-friendly way, with warnings about the feature being required to be able to use this part of the code.

`doc_cfg` is in the process of being stabilized, a RFC is being written, but nobody has a date for now

Reply:
> I know this is not meant to be a definitive version

Actually, it could be merged now, as everything is there and working. It will just depends on your feedbacks now

Reply:
Oh, then I'll review it again later.

Reply:
I added a comment with some explainations